diff -up libetpan/src/low-level/imap/namespace_parser.c.orig libetpan/src/low-level/imap/namespace_parser.c
--- libetpan/src/low-level/imap/namespace_parser.c.orig	2011-03-12 09:13:32.000000000 +0100
+++ libetpan/src/low-level/imap/namespace_parser.c	2011-03-12 09:14:01.000000000 +0100
@@ -29,12 +29,11 @@
  * SUCH DAMAGE.
  */
 
-#include "namespace_parser.h"
-
 #ifdef HAVE_CONFIG_H
 #	include <config.h>
 #endif
 
+#include "namespace_parser.h"
 #include "namespace_types.h"
 #include "mailimap_keywords.h"
 #include "mailimap_extension.h"
